state-synchronizer/src/request_manager.rs

```diff
@@ -456,12 +456,17 @@ impl RequestManager {
         let multicast_start_time = self.get_multicast_start_time(version).unwrap_or(UNIX_EPOCH);
         if is_timeout(multicast_start_time, self.multicast_timeout) {
             let new_multicast_level = std::cmp::min(
-                self.multicast_level + 1,
-                self.upstream_config.upstream_count() - 1, // multicast_level (=network preference) is 0-indexed
+                self.multicast_level
+                    .checked_add(1)
+                    .ok_or_else(|| format_err!("New multicast level has overflown!"))?,
+                self.upstream_config
+                    .upstream_count()
+                    .checked_sub(1)
+                    .ok_or_else(|| format_err!("Upstream count has overflown!"))?, // multicast_level (=network preference) is 0-indexed
```

Comment:
if the only case where we call this is to default to false on error, should we return `false` instead an of an error here?

Reply:
Similar to the above, I'd prefer to make this "future proof" in the sense that I'd rather not hide the error but make the caller handle it. I worry that if we indeed hit over/underflows, it's something that should be flagged immediately.